tests/test_pair_tcp.cpp

```diff
@@ -29,23 +29,39 @@
 
 #include "testutil.hpp"
 
-int main (void)
+typedef void (*extra_func_t) (void *socket);
+
+void set_sockopt_fastpath (void *socket)
+{
+    int value = 1;
+    int rc =
+      zmq_setsockopt (socket, ZMQ_LOOPBACK_FASTPATH, &value, sizeof value);
```

Comment:
won't this fail on non-windows?

Reply:
Not like I implemented it at the moment. It just won't have an effect. This will also be the case if it is used on a Windows version that does not support it. This behaviour could be changed, and an error be returned on non-Windows systems (but the same option could be used for similar optimizations on other OS's if present), or also on Windows systems that don't support it. However, this would require error handling by the caller, e.g. when using with zmqpp, this would throw an exception, which I found suboptimal. With the current design, the option can always be set, regardless of the execution environment, as long as the caller ensures that both sides of the connection apply it. This must still be documented, of course!

Would you do this differently?

Reply:
Gah, I'm just very tired and for some reasons I read setsockopt rather than zmq_setsockopt, sorry.

Although there are other options that are platform specific if you check in src/options.cpp, and they are compiled out which means they return EINVAL, so perhaps it's worth keeping it consistent?

Reply:
```
../tests/test_pair_tcp.cpp: In function ‘void set_sockopt_fastpath(void*)’:
../tests/test_pair_tcp.cpp:38:31: error: ‘ZMQ_LOOPBACK_FASTPATH’ was not declared in this scope
       zmq_setsockopt (socket, ZMQ_LOOPBACK_FASTPATH, &value, sizeof value);
```

It also needs to be ifdefd for drafts

Reply:
Regarding EINVAL: I agree if we regard this as platform-specific. However, I somehow see this particular option as platform-agnostic in the following sense: Setting ZMQ_LOOPBACK_FASTPATH to 1 can be done on any platform to activate any optimizations for TCP loopback connections that platform provides which need to be activated explicitly on both connection ends. On platforms that do not provide such optimizations, or enable them by default, this will be a no-op.

This would mean that a cross-platform app could set this option on any platform. Otherwise, it would need to make some platform-specific distinction.

However, on Windows, I would return success regardless of whether the actual OS version supports it or not. The actual result of setting the underlying sockopt is not available at that point, and so we would need to somehow "guess" if it will be successful.

If you still think EINVAL is better on non-Windows, I am happy to change it :)

Regarding the DRAFT: Will fix that.

Reply:
Ok that's fine, let's leave it

Reply:
Fixed non-DRAFT